File: Zadanie2/Zadanie2-SOLID/Zadanie2-SOLID/Master.cs
namespace Zadanie2_SOLID
{
    class Master
    {
        private List<IPerson> ListOfPersons { get; set; } = [];

        public void Show()
        {
            foreach (IPerson inihabitant in ListOfPersons)
            {
                Console.WriteLine(inihabitant.Describe());
            }
        }

        public void Play()
        {
            for (int i = 0; i < 10; i++)
            {
                Random random = new();
                IPerson inhabitant = this.ListOfPersons[random.Next(this.ListOfPersons.Count)];
                Console.WriteLine(inhabitant.Describe());

                while (random.NextDouble() < 0.75)
                {
                    Console.WriteLine(inhabitant.Play());
                }
            }
        }

        public void SetInhabitants()
        {
            this.ListOfPersons.Add(new Inhabitant("Slenderwillow", "f", "Oldoak", "Sunnydaisy", new Gardener()));
            this.ListOfPersons.Add(new Inhabitant("Quickhand", "m", "Bravedeed", "Sunbell", new Knight("sword", "horse")));
            this.ListOfPersons.Add(new Inhabitant("Truefriend", "m", "Highspirit", "Mistymorning", new Knight("spear", "dragon")));
            this.ListOfPersons.Add(new Inhabitant("Greenleaf", "m", "Brownleaf", "Goldendaisy", new Gardener()));
        }

        public void ChangeInhabitantRole(int personIndex, IRole role)
        {
            if (personIndex < 0 || personIndex >= this.ListOfPersons.Count)
            {
                Console.WriteLine("Incorrect index.");
            }
            else
            {
                IPerson inhabitant = this.ListOfPersons[personIndex];
                inhabitant.ChangeRole(role);
            }
        }
    }
}

File: Zadanie2-SOLID/Polecenie/solid_2/solid_2_py/master.py
# -*- coding: utf-8 -*-
"""
Created on Fri Nov 29 09:57:32 2019

@author: Ewa
"""


from random import choice, uniform
from gardener import Gardener
from knight import Knight


class Master:
    def __init__(self):
        self.list_of_persons = []

    def show(self):
        for p in self.list_of_persons:
            print(p.describe())

    def play(self):
        for i in range(10):
            person = choice(self.list_of_persons);
            print(person.describe())
            while uniform(0, 1) < 0.5:
                if person.get_role() == 'Knight':
                    print(person.fight())
                else:
                    print(person.plant())

    def set_inhabitants(self):
        self.list_of_persons.append(Gardener('Greenleaf','m','Brownleaf','Goldendaisy'));
        self.list_of_persons.append(Gardener('Slenderwillow','f','Oldoak','Sunnydaisy'));
        self.list_of_persons.append(Knight('Quickhand','m','Bravedeed',
                'Sunbell','sword','horse'));
        self.list_of_persons.append(Knight('Truefriend','m','Highspirit',
                'Mistymorning','spear','dragon'));


def main():
    m = Master()
    m.set_inhabitants()
    m.play()

if __name__ == '__main__':
    main()

